src/validation.cpp

```diff
@@ -4189,7 +4189,7 @@ bool ChainstateManager::ProcessNewBlockHeaders(const std::vector<CBlockHeader>&
         if (IsInitialBlockDownload() && ppindex && *ppindex) {
             const CBlockIndex& last_accepted{**ppindex};
             const int64_t blocks_left{(GetTime() - last_accepted.GetBlockTime()) / GetConsensus().nPowTargetSpacing};
-            const double progress{100.0 * last_accepted.nHeight / (last_accepted.nHeight + blocks_left)};
+            const double progress{100.0 * (last_accepted.nHeight + 1) / (last_accepted.nHeight + 1 + blocks_left)};
```

Comment:
If `blocks_left` can be negative, it doesn't look like the fix really solves the issue. `blocks_left` might as well be `-2` instead of `-1`, and we divide by zero again. Might be better to check if it's zero and, if so, skip the log?

Reply:
I'll remove it from here and rebase on top of that once merged